src/Elastic.Apm.AspNetCore/ApmMiddleware.cs

```diff
@@ -78,12 +80,50 @@ public async Task InvokeAsync(HttpContext context)
 					ApiConstants.TypeRequest);
 			}
 
+			if (transaction.IsSampled) FillSampledTransactionContextRequest(context, transaction);
+
+			try
+			{
+				await _next(context);
+			}
+			catch (Exception e) when (ExceptionFilter.Capture(e, transaction)) { }
+			finally
+			{
+				//fixup Transaction.Name - e.g. /user/profile/1 -> /user/profile/{id}
+				var routeData = (context.Features[typeof(IRoutingFeature)] as IRoutingFeature)?.RouteData;
+				if (routeData != null)
+				{
+					var name = GetNameFromRouteContext(routeData.Values);
+
+					if (!string.IsNullOrWhiteSpace(name)) transaction.Name = $"{context.Request.Method} {name}";
+				}
+
+				transaction.Result = Transaction.StatusCodeToResult(GetProtocolName(context.Request.Protocol), context.Response.StatusCode);
+
+				if (transaction.IsSampled)
+				{
+					FillSampledTransactionContextResponse(context, transaction);
+					FillSampledTransactionContextUser(context, transaction);
+				}
+
+				transaction.End();
+			}
+		}
+
+		private string GetRawUrl(HttpRequest httpRequest)
+		{
+			var rawPathAndQuery = httpRequest.HttpContext.Features.Get<IHttpRequestFeature>()?.RawTarget;
+			return rawPathAndQuery == null ? null : UriHelper.BuildAbsolute(httpRequest.Scheme, httpRequest.Host, rawPathAndQuery);
+		}
+
+		private void FillSampledTransactionContextRequest(HttpContext context, Transaction transaction)
+		{
 			var url = new Url
 			{
-				Full = context.Request?.Path.Value,
+				Full = context.Request?.GetEncodedUrl(),
 				HostName = context.Request.Host.Host,

```

Comment:
Could we also add a `?` after `Request`? Like `HostName = context.Request?.Host.Host,`

Reply:
Done

Reply:
👍 Perfect! I accidentally clicked on sending that, while I was not through the whole change set. That's exactly what my conclusion was by looking at the next lines.